demos/python_demos/bert_question_answering/question_answering_demo.py

```diff
@@ -0,0 +1,338 @@
+def get_context(args):
+    log.info("Get context from {}".format(args.input))
+    response = requests.get(args.input)
+    if response.status_code == 404:
+        raise Exception('URL is invalid')
+    html = bs4.BeautifulSoup(response.text, 'html.parser')
+    heads = html.select("#firstHeading")
+    title = heads[0].text if heads else "no title"
+    paragraphs = html.select("p")
+    log.info("Article title: '{}'".format(title))
+    context = '\n'.join([par.text for par in paragraphs])
+    log.info("Size: {} chars".format(len(context)))
+    log.info("Context: \033[91m {} \033[0m".format(context))
+    return context
+
+
+def main():
+    log.basicConfig(format="[ %(levelname)s ] %(message)s", level=log.INFO, stream=sys.stdout)
+    args = build_argparser().parse_args()
+
+    # load vocabulary file for model
+    log.info("Loading vocab file:\t{}".format(args.vocab))
+    with open(args.vocab, "r", encoding="utf-8") as r:
+        vocab = dict((t.rstrip("\n"), i) for i, t in enumerate(r.readlines()))
+    log.info("{} tokens loaded".format(len(vocab)))
+
+    # get context as a string (as we might need it's length for the sequence reshape)
+    context = get_context(args)
+    # encode context into token ids list
+    c_tokens_id, c_tokens_se = text_to_tokens(context, vocab)
+
+    log.info("Initializing Inference Engine")
+    ie = IECore()
+    log.info("Device is {}".format(args.device))
+    version = ie.get_versions(args.device)[args.device]
+    version_str = "{}.{}.{}".format(version.major, version.minor, version.build_number)
+    log.info("Plugin version is {}".format(version_str))
+
+    # read IR
+    model_xml = args.model
+    model_bin = os.path.splitext(model_xml)[0] + ".bin"
+    log.info("Loading network files:\n\t{}\n\t{}".format(model_xml, model_bin))
+    ie_encoder = ie.read_network(model=model_xml, weights=model_bin)
+
+    if args.reshape:
+        # reshape the sequence length to the context + maximum question length (in tokens)
+        first_input_layer = next(iter(ie_encoder.inputs))
+        c = ie_encoder.inputs[first_input_layer].shape[1]
+        # find the closest multiple of 64
+        seq = min(c, round((len(c_tokens_id) + args.max_question_token_num) / 64) * 64)
+        if seq < c:
+            input_info = list(ie_encoder.inputs)
+            new_shapes = dict([])
+            for i in input_info:
+                n, c = ie_encoder.inputs[i].shape
+                new_shapes[i] = [n, seq]
+                log.info("Reshaped input {} from {} to the {}".format(i, ie_encoder.inputs[i].shape, new_shapes[i]))
+            log.info("Attempting to reshape the network to the modified inputs...")
+            try:
+                ie_encoder.reshape(new_shapes)
+                log.info("Successful!")
+            except:
+                log.info("Failed...reloading the network")
+                ie_encoder = ie.read_network(model=model_xml, weights=model_bin)
```

Comment:
That approach with reloading the net would make scene if there was no demo key managing that. I think it is simpler and clearer just to fail if the model can't be reshaped. So next time a user just shouldn't request reshaping.